src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PointerElementAccess.cs

```diff
@@ -12,12 +8,24 @@ internal sealed partial class LocalRewriter
     {
         public override BoundNode VisitPointerElementAccess(BoundPointerElementAccess node)
         {
-            BoundExpression rewrittenExpression = VisitExpression(node.Expression);
+            BoundExpression rewrittenExpression = LowerReceiverOfPointerElementAccess(node.Expression);
             BoundExpression rewrittenIndex = VisitExpression(node.Index);
 
             return RewritePointerElementAccess(node, rewrittenExpression, rewrittenIndex);
         }
 
+        private BoundExpression LowerReceiverOfPointerElementAccess(BoundExpression receiver)
+        {
+            if (receiver is BoundFieldAccess fieldAccess && fieldAccess.FieldSymbol.IsFixed)
+            {
+                var loweredFieldReceiver = VisitExpression(fieldAccess.ReceiverOpt);
+                fieldAccess.Update(loweredFieldReceiver, fieldAccess.FieldSymbol, fieldAccess.ConstantValueOpt, fieldAccess.ResultKind, fieldAccess.Type);
```

Comment:
`Update` isn't side-effecting, is it?
I suspect we don't need this `Update`, since we're discarding the result anyways...